editoast/src/views/path/properties.rs

```diff
@@ -52,6 +58,57 @@ editoast_common::schemas! {
 pub struct PathPropertiesInput {
     /// List of track sections
     pub track_section_ranges: Vec<TrackRange>,
+    /// The path offset in mm of each path item given as input of the pathfinding
+    /// The first value is always `0` (beginning of the path) and the last one is always equal to the `length` of the path in mm
+    pub path_item_positions: Vec<u64>,
+}
+
+impl PathPropertiesInput {
+    fn get_track_length_cumulative_sums(&self) -> Vec<u64> {
+        let mut cumulative_sums = Vec::new();
+        let mut cumulative_sum = 0;
+
+        for track_section in self.track_section_ranges.iter() {
```

Comment:
nitpick: this can be written like this (not strictly the same but with equivalent behavior, iteration on references of the `Item`)

```suggestion
        for track_section in &self.track_section_ranges {
```